OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorSelectionLogic.cs

```diff
@@ -31,21 +32,29 @@ public class MapEditorSelectionLogic : ChromeLogic
 		readonly CheckboxWidget copyActorsCheckbox;
 		readonly EditorActorLayer editorActorLayer;
 
+		public LabelWidget RegionLabel;
+		public LabelWidget DimensionsLabel;
+		public LabelWidget DiagonalLabel;
+		public LabelWidget ResourceCounterLabel;
+
 		MapCopyFilters copyFilters = MapCopyFilters.All;
 		EditorClipboard? clipboard;
 
 		readonly IResourceLayer resourceLayer;
 
+		readonly EditorResourceLayer editorResourceLayer;
+
 		[ObjectCreator.UseCtor]
 		public MapEditorSelectionLogic(Widget widget, World world, WorldRenderer worldRenderer)
 		{
 			this.worldRenderer = worldRenderer;
 
 			editorActorLayer = world.WorldActor.Trait<EditorActorLayer>();
 			resourceLayer = world.WorldActor.Trait<IResourceLayer>();
+			editorResourceLayer = world.WorldActor.Trait<EditorResourceLayer>();
 
 			editor = widget.Get<EditorViewportControllerWidget>("MAP_EDITOR");
-
+			editor.DefaultBrush.SelectionChanged += HandleSelectionChanged;
```

Comment:
If you subscribe to an event you should also unsubscribe. That is usually done is a Disposed() method

Reply:
Whre should unsubscribe be done?

Reply:
```csharp
protected override void Dispose(bool disposing)
{
	editor.DefaultBrush.SelectionChanged -= HandleSelectionChanged;
	base.Dispose(disposing);
}
```

Reply:
thx